### ISSUE TITLE
Add AppSet webhook endpoint in argocd-server

### ISSUE THREAD

**Issue by Syst3m1cAn0maly** [argoproj/argo-cd#13764]
# Summary

Add a webhook endpoint in argocd-server to refresh AppSets

# Motivation

Currently you have to add an Ingress or a NodePort to expose the webhook endpoint from the applicationset-controller. It is not practical and not present in the argohelm chart...

# Proposal

We could add an endpoint on argocd-server for example /api/applicationset/webhook and reverse proxy to the applicationset-controller based on what is already implemented for the dex server

**Comment by mubarak-j** [comment]
it's currently possible to define the AppSet webhook using the helm-chart [here](https://github.com/argoproj/argo-helm/blob/438f7a26b7518ec1fc4133f12f58cb0b8d1a2765/charts/argo-cd/values.yaml#L2588-L2631)

**Comment by Syst3m1cAn0maly** [comment]
@mubarak-j 
You are right, you can define it with the helm-chart, however it is limited to ingress configuration, you can't expose it with a NodePort service for example.
From the architectural standpoint, it makes a lot of sense that the only endpoint exposed is the argocd-server, and the applicationset-controller is currently an exception to that...

**Comment by pre** [comment]
Related: Provide a single entrypoint for all public webhooks  https://github.com/argoproj/argo-cd/issues/14954

**Comment by Kishore1705** [comment]
now that i have created a route to ApplicationSet controller service , should i add it as Webhook anywhere?
